tests/basic_cxx_qt/cpp/main.cpp

```diff
@@ -5,225 +5,222 @@
+    QSignalSpy float32Spy(&types, &cxx_qt::my_types::MyTypes::float32Changed);
+    QSignalSpy float64Spy(&types, &cxx_qt::my_types::MyTypes::float64Changed);
+    QSignalSpy int8Spy(&types, &cxx_qt::my_types::MyTypes::int8Changed);
+    QSignalSpy int16Spy(&types, &cxx_qt::my_types::MyTypes::int16Changed);
+    QSignalSpy int32Spy(&types, &cxx_qt::my_types::MyTypes::int32Changed);
+    QSignalSpy uint8Spy(&types, &cxx_qt::my_types::MyTypes::uint8Changed);
+    QSignalSpy uint16Spy(&types, &cxx_qt::my_types::MyTypes::uint16Changed);
+    QSignalSpy uint32Spy(&types, &cxx_qt::my_types::MyTypes::uint32Changed);
+
+    QCOMPARE(types.getBoolean(), false);
+    QCOMPARE(booleanSpy.count(), 0);
+    types.setBoolean(true);
+    QCOMPARE(booleanSpy.count(), 0);
+    QCoreApplication::processEvents();
+    QCOMPARE(booleanSpy.count(), 1);
+    QCOMPARE(types.getBoolean(), true);
+
+    QCOMPARE(types.getFloat32(), 0.0);
+    QCOMPARE(float32Spy.count(), 0);
+    types.setFloat32(0.33f);
+    QCOMPARE(float32Spy.count(), 0);
+    QCoreApplication::processEvents();
+    QCOMPARE(float32Spy.count(), 1);
+    QCOMPARE(types.getFloat32(), 0.33f);
+
+    QCOMPARE(types.getFloat64(), 0.0);
+    QCOMPARE(float64Spy.count(), 0);
+    types.setFloat64(0.33);
+    QCOMPARE(float64Spy.count(), 0);
+    QCoreApplication::processEvents();
+    QCOMPARE(float64Spy.count(), 1);
+    QCOMPARE(types.getFloat64(), 0.33);
+
+    QCOMPARE(types.getInt8(), 0);
+    QCOMPARE(int8Spy.count(), 0);
+    types.setInt8(4);
+    QCOMPARE(int8Spy.count(), 0);
+    QCoreApplication::processEvents();
+    QCOMPARE(int8Spy.count(), 1);
+    QCOMPARE(types.getInt8(), 4);
+
+    QCOMPARE(types.getInt16(), 0);
+    QCOMPARE(int16Spy.count(), 0);
+    types.setInt16(4);
+    QCOMPARE(int16Spy.count(), 0);
+    QCoreApplication::processEvents();
+    QCOMPARE(int16Spy.count(), 1);
+    QCOMPARE(types.getInt16(), 4);
+
+    QCOMPARE(types.getInt32(), 0);
+    QCOMPARE(int32Spy.count(), 0);
+    types.setInt32(4);
+    QCOMPARE(int32Spy.count(), 0);
+    QCoreApplication::processEvents();
+    QCOMPARE(int32Spy.count(), 1);
+    QCOMPARE(types.getInt32(), 4);
+
+    QCOMPARE(types.getUint8(), 0);
+    QCOMPARE(uint8Spy.count(), 0);
+    types.setUint8(4);
+    QCOMPARE(uint8Spy.count(), 0);
+    QCoreApplication::processEvents();
+    QCOMPARE(uint8Spy.count(), 1);
+    QCOMPARE(types.getUint8(), 4);
+
+    QCOMPARE(types.getUint16(), 0);
+    QCOMPARE(uint16Spy.count(), 0);
+    types.setUint16(4);
+    QCOMPARE(uint16Spy.count(), 0);
+    QCoreApplication::processEvents();
+    QCOMPARE(uint16Spy.count(), 1);
+    QCOMPARE(types.getUint16(), 4);
+
+    QCOMPARE(types.getUint32(), 0);
+    QCOMPARE(uint32Spy.count(), 0);
+    types.setUint32(4);
+    QCOMPARE(uint32Spy.count(), 0);
+    QCoreApplication::processEvents();
+    QCOMPARE(uint32Spy.count(), 1);
+    QCOMPARE(types.getUint32(), 4);
+  }
+};
+
+QTEST_MAIN(CxxQtTest)
+#include "main.moc"
```

Comment:
Is this line required?

Reply:
Yes otherwise automoc fails. It's the common way of doing it for source only files like a main.cpp.

Reply:
Eg this is the output
```
[build] "SRC:/tests/basic_cxx_qt/cpp/main.cpp"
[build] contains a "Q_OBJECT" macro, but does not include "main.moc"!
[build] Consider to
[build]   - add #include "main.moc"
[build]   - enable SKIP_AUTOMOC for this file
```

Then if you remove the `Q_OBJECT` macro, none of the tests run as QTest expects private Q_SLOTS to be tests. But as it's not then a Q_OBJECT it doesn't know there are slots.

Reply:
And this is actually what they do in the QTest tutorial as well :-) https://doc.qt.io/qt-6/qttestlib-tutorial1-example.html#preparing-the-stand-alone-executable